Xamarin.Forms.Platform.Android/CollectionView/CarouselViewRenderer.cs

```diff
@@ -370,7 +395,18 @@ void UpdateVisualStates()
 		void CarouselViewScrolled(object sender, ItemsViewScrolledEventArgs e)
 		{
 			_noNeedForScroll = false;
-			UpdatePosition(e.CenterItemIndex);
+			var index = e.CenterItemIndex;
+			if (Carousel.Loop)
+			{
+				var centeredView = this.GetCenteredView();
+
+				if (centeredView is ItemContentView templatedCell)
+				{
+					var bContext = templatedCell?.Element?.BindingContext;
+					index = ItemsViewAdapter.GetPositionForItem(bContext);
```

Comment:
Once you have the `centeredView`, can you just get the position using [getChildAdapterPosition](https://developer.android.com/reference/kotlin/androidx/recyclerview/widget/RecyclerView#getchildadapterposition)?

Reply:
because if the adapter is on a loop the position on the adapter is not the real position of the item in the ItemSource the user gave us.